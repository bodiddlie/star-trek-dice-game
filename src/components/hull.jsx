import React from 'react';
import { connect } from 'react-redux';

export const Hull = ({ hull }) =>
  <div>
    {hull}
  </div>;

const mapStateToProps = state => ({
  hull: state.hull,
});

export default connect(mapStateToProps)(Hull);
