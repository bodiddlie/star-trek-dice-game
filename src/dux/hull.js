// @flow
import { type Action } from './types';

const TAKE_DAMAGE = '[Hull] Take Damage';
const REPAIR_DAMAGE = '[Hull] Repair Damage';
const RESET_NEGATIVE_HULL = '[Hull] Reset Negative Hull';

export default function reducer(state: number = 10, action: Action): number {
  switch (action.type) {
    case TAKE_DAMAGE: {
      return state - action.payload;
    }
    case REPAIR_DAMAGE: {
      return Math.min(10, state + action.payload);
    }
    case RESET_NEGATIVE_HULL: {
      return state <= 0 ? 0 : state;
    }
    default: {
      return state;
    }
  }
}

export function takeDamage(amount: number): Action {
  return { type: TAKE_DAMAGE, payload: amount };
}

export function repairDamage(amount: number): Action {
  return { type: REPAIR_DAMAGE, payload: amount };
}

export function resetNegativeHull(): Action {
  return { type: RESET_NEGATIVE_HULL, payload: null };
}
