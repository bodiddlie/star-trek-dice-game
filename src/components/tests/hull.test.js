import React from 'react';
import { shallow } from 'enzyme';

import { Hull } from '../hull';

describe('<Hull />', () => {
  it('should render the hull value', () => {
    const wrapper = shallow(<Hull hull={7} />);
    expect(wrapper).toMatchSnapshot();
  });
});
