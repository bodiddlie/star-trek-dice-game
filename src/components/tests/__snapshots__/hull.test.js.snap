// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hull /> should render the hull value 1`] = `
<div>
  7
</div>
`;
